staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -219,8 +219,7 @@ func NamespaceKeyRootFunc(ctx context.Context, prefix string) string {
 // context does not contain a namespace, it errors.
 func NamespaceKeyFunc(ctx context.Context, prefix string, name string) (string, error) {
 	key := NamespaceKeyRootFunc(ctx, prefix)
-	ns, ok := genericapirequest.NamespaceFrom(ctx)
-	if !ok || len(ns) == 0 {
+	if key == prefix {
```

Comment:
this depends on the error behavior of the NamespaceKeyRootFunc function, which is not desired. if we want to move the key computation after the NamespaceFrom check, that would be ok.